### PR TITLE
Feature/rename without firebase

### PR DESCRIPTION
**Pull request by achocron**

### CODE REVIEW

File: sdk/appcenter-push/src/main/java/com/microsoft/appcenter/push/Push.java [hunk 1]
```diff
@@ -20,41 +20,20 @@
...
-    };
-
-    /**
      * Name of the service.
      */
     private static final String SERVICE_NAME = "Push";
 
     /**
-     * TAG used in logging for Analytics.
+     * TAG used in logging for Push.
```
Comment (MatkovIvan) [inline sdk/appcenter-push/src/main/java/com/microsoft/appcenter/push/Push.java]: Good catch!

File: sdk/appcenter-push/src/main/java/com/microsoft/appcenter/push/Push.java [hunk 2]
```diff
@@ -107,6 +112,9 @@ private Push() {
     @SuppressWarnings("WeakerAccess")
     public static synchronized Push getInstance() {
         if (sInstance == null) {
+
+            /* Explicitly set to false for testing. */
+            sFirebaseAnalyticsEnabled = false;
```
Comment (guperrot) [inline sdk/appcenter-push/src/main/java/com/microsoft/appcenter/push/Push.java]: Use a member variable to avoid this

File: sdk/appcenter-push/src/main/java/com/microsoft/appcenter/push/Push.java [hunk 3]
```diff
@@ -247,6 +283,11 @@ protected int getTriggerCount() {
     public synchronized void onStarted(@NonNull Context context, @NonNull String appSecret, @NonNull Channel channel) {
         mContext = context;
         super.onStarted(context, appSecret, channel);
+
+        if (FirebaseUtils.isFirebaseAvailable() && !sFirebaseAnalyticsEnabled) {
```
Comment (guperrot) [inline sdk/appcenter-push/src/main/java/com/microsoft/appcenter/push/Push.java]: either add a comment or remove blank line

File: sdk/appcenter-push/src/main/java/com/microsoft/appcenter/push/Push.java [hunk 4]
```diff
@@ -177,6 +185,34 @@ private synchronized void instanceSetSenderId(String senderId) {
         mSenderId = senderId;
     }
 
+    /**
+     * Enable firebase analytics collection.
+     *
+     * @param context the context to retrieve FirebaseAnalytics instance.
+     */
+    @Deprecated
```
Comment (guperrot) [inline sdk/appcenter-push/src/main/java/com/microsoft/appcenter/push/Push.java]: Also use the Javadoc deprecated tag.

File: sdk/appcenter-push/src/main/java/com/microsoft/appcenter/push/Push.java [hunk 5]
```diff
@@ -183,7 +183,10 @@ private synchronized void instanceSetSenderId(String senderId) {
     }
 
     /**
-     * @deprecated
+     * @deprecated Future releases will not alter the state of Firebase Analytics data collection by
```
Comment (guperrot) [inline sdk/appcenter-push/src/main/java/com/microsoft/appcenter/push/Push.java]: Please commit (not push but at least commit) using Android Studio and enable this:

screen shot 2017-11-17 at 2 17 31 pm

As for this particular block, please format the file again.

Comment (achocron) [reply sdk/appcenter-push/src/main/java/com/microsoft/appcenter/push/Push.java]: What should the format be?

**Comment by guperrot** [comment]
@achocron We will need to prepare the doc changes before we can merge.
Portal doc changes should be behind a feature flag otherwise we can't really be in sync between portal deployment and SDK release (or we need to release a tuesday right before 2pm).

**Comment by guperrot** [comment]
Please also address these lint issues:

```

File: apps/sasquatch/src/main/java/com/microsoft/appcenter/sasquatch/activities/SettingsActivity.java [hunk 6]
```diff
@@ -37,7 +37,6 @@
 
 import static com.microsoft.appcenter.sasquatch.activities.MainActivity.APP_SECRET_KEY;
 import static com.microsoft.appcenter.sasquatch.activities.MainActivity.FILE_ATTACHMENT_KEY;
-import static com.microsoft.appcenter.sasquatch.activities.MainActivity.FIREBASE_ENABLED_KEY;
```
Comment (guperrot) [inline apps/sasquatch/src/main/java/com/microsoft/appcenter/sasquatch/activities/SettingsActivity.java]: Why removing that since we kept the API it's still useful in sasquatch to test

File: sdk/appcenter/src/test/java/com/microsoft/appcenter/utils/AppNameHelperTest.java [hunk 7]
```diff
@@ -0,0 +1,47 @@
+package com.microsoft.appcenter.utils;
+
+
```
Comment (guperrot) [inline sdk/appcenter/src/test/java/com/microsoft/appcenter/utils/AppNameHelperTest.java]: Remove 1 blank line

File: sdk/appcenter/src/test/java/com/microsoft/appcenter/utils/AppNameHelperTest.java [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+        String appName = "non-localized-app-name";
+        Context context = mock(Context.class);
+        ApplicationInfo applicationInfo = mock(ApplicationInfo.class);
+        when(context.getApplicationInfo()).thenReturn(applicationInfo);
+        Whitebox.setInternalState(applicationInfo, "labelRes", 0);
+        Whitebox.setInternalState(applicationInfo, "nonLocalizedLabel", appName);
+        String retrievedAppName = AppNameHelper.getAppName(context);
+        assertEquals(appName, retrievedAppName);
+    }
+
```
Comment (guperrot) [inline sdk/appcenter/src/test/java/com/microsoft/appcenter/utils/AppNameHelperTest.java]: blank line not needed here

File: sdk/appcenter-push/src/main/java/com/microsoft/appcenter/push/Push.java [hunk 9]
```diff
@@ -202,15 +229,17 @@ private void enqueuePushInstallationLog(@NonNull String pushToken) {
...
-    synchronized void onTokenRefresh(@NonNull final String pushToken) {
-        AppCenterLog.debug(LOG_TAG, "Push token refreshed: " + pushToken);
-        post(new Runnable() {
-
-            @Override
-            public void run() {
-                enqueuePushInstallationLog(pushToken);
-            }
-        });
+    synchronized void onTokenRefresh(final String pushToken) {
```
Comment (jwallra) [inline sdk/appcenter-push/src/main/java/com/microsoft/appcenter/push/Push.java]: What's the reason for the change here, moving the non-null check inside the function?

Comment (guperrot) [reply sdk/appcenter-push/src/main/java/com/microsoft/appcenter/push/Push.java]: So that outside calls don't have to check
